### ISSUE TITLE
SnippetAPI.ListSnippets is unable to properly unmarshal the JSON response from Fastly

### ISSUE THREAD

**Issue by kevin-fagan** [fastly/fastly-go#4]
**Version**
v1.0.0-beta.10

**What happened**

According to the [Fastly Documentation](https://developer.fastly.com/reference/api/vcl-services/snippet/) for the snippet API, the `dynamic` field is an `integer`. However, when you look at the actual HTTP response from the Fastly REST API, the `dynamic` field comes through as a `string`. See below:

```json
[
  {
    "priority": "700",
    "type": "error",
    "service_id": "id",
    "deleted_at": null,
    "id": "id",
    "created_at": "2023-02-21T19:43:11Z",
    "updated_at": "2023-06-07T03:28:57Z",
    "dynamic": "0",
    "version": "303",
    "content": "content",
    "name": "name"
  },
]
```
**So why is this an issue?**
When you attempt to retrieve all snippets that belong to a Fastly Service using the `SnippetAPI.ListSnippets` function, it returns nil values for the slice of `fastly.SnippetResponses`.  Behind the scenes, when the function attempts to unmarshal the JSON payload, it is unable to properly unmarshal the `dynamic` field which causes the function to return a slice of nil values.

**Comment by Integralist** [comment]
Hi @kevin-fagan 

Thanks for opening this issue. Our API clients are auto-generated from our OpenAPI specifications ([blog post](https://dev.to/fastly/better-fastly-api-clients-with-openapi-generator-3lno)) and we've become aware of a discrepancy between what the API returns and what's documented (which of course has an effect on the generated API client code).

I've now raised this internally with the relevant team. 

For now, I'll leave this issue open and once I have feedback I'll provide it here.

Thanks again,

**Comment by Integralist** [comment]
👋🏻 

This has now been fixed in the `v1.0.0-beta.16` release.

Example:

```go
package main

import (
	"fmt"
	"os"

	"github.com/fastly/fastly-go/fastly"
)

func main() {
	cfg := fastly.NewConfiguration()
	apiClient := fastly.NewAPIClient(cfg)
	ctx := fastly.NewAPIKeyContextFromEnv("FASTLY_API_TOKEN")

	serviceID := "<SERVICE_ID>"
	versionID := int32(<VERSION>)

	resp, r, err := apiClient.SnippetAPI.ListSnippets(ctx, serviceID, versionID).Execute()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error when calling `SnippetAPI.ListSnippets`: %v\n", err)
		fmt.Fprintf(os.Stderr, "Full HTTP response: %v\n", r)
	}

	for _, snippet := range resp {
		fmt.Printf("snippet: %+v | dynamic: %+v\n", *snippet.Name, *snippet.Dynamic)
	}
}
```

Output:

```
snippet: bug-testing | dynamic: 0
snippet: bug-testing-dynamic | dynamic: 1
```
